tests/test_openPathUpdateAll.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+from typing import Tuple
+
+##### Needed for importing script files (as opposed to classes)
+import sys
+import os
+sys.path.insert(0, os.path.dirname(os.path.dirname(__file__)))
+##### End block
+
+import openPathUpdateAll
+from mock_neon_users import MockNeonUserBuilder
+from mock_alta_users import MockAltaUserBuilder
+
+class TestOpenPathUpdateAll:
+    def _create_alta_accounts(self, *group_lists):
+        accts = {}
+        for i, groups in enumerate(group_lists):
+            # Cannot have a user with id 0
+            acct = (MockAltaUserBuilder()
+                    .with_id(i+1)
+                    .with_groups(groups)
+                    .build()
+                   )
+            accts[acct['OpenPathID']] = acct
+        return accts
+
+    def _create_neon_accounts(self, alta_ids):
+        accts = {}
+        for aid in alta_ids:
+            acct = (MockNeonUserBuilder()
+                    .with_alta_id(aid)
+                    .build())
+            accts.update({acct['id']: acct})
+        return accts
+
+    def _create_matching_alta_and_neon_accounts(self, group_lists):
+        # Create two users with accounts in Alta and in Neon (linked by OpenPathID)
+        alta_accounts = self._create_alta_accounts(*group_lists)
+        neon_accounts = self._create_neon_accounts(alta_accounts.keys())
+        return (alta_accounts, neon_accounts)
+
+
+    @pytest.fixture
+    def setup_mocks(self, mocker):
```

Comment:
How about mocking requests to throw an exception, just to doubly ensure that the tests don't run any network requests & everything is mocked properly? You could do this with a conftest.py like this:

```
import pytest
from unittest.mock import patch


@pytest.fixture(autouse=True)
def _no_requests():
    with patch(
        "requests.sessions.Session.request",
        side_effect=RuntimeError("Network disabled in tests"),
    ):
        yield
```